### PR TITLE
Feat/search tree

### LINKED ISSUE #21: Implement search in TMLP

### PR DESCRIPTION
**Pull request by mari1912**
## Description

- Add search to tree list page

## Related Issues
- Closes #21

### CODE REVIEW

**Comment by lfjnascimento** [comment]
It would be great to have debouncing so that the component doesn't re-render/recalculate data on every keystroke

File: dashboard/src/components/TopBar/TopBar.tsx [hunk 1]
```diff
@@ -12,7 +18,7 @@ const TopBar = (): JSX.Element => {
         <div className="flex w-2/3 px-6 items-center">
           {/* placeholder for search */}
           {/* TODO: use i18n for the input placeholder */}
```
Comment (viniciuslacerda0) [inline dashboard/src/components/TopBar/TopBar.tsx:20]: is this to-do still necessary?

File: dashboard/src/components/TreeListingPage/TreeListingPage.tsx [hunk 2]
```diff
@@ -66,19 +70,21 @@ const FilterButton = (): JSX.Element => {
...
 
-const TreeListingPage = (): JSX.Element => {
+const TreeListingPage = ({inputFilter}: ITreeListingPage): JSX.Element => {
 
   const { data } = useTreeTable();
 
   const listItems: TreeTableBody[] = useMemo(() => {
-    return data ? (data as Tree[]).map(tree => ({
+    return data ? (data as Tree[])
+    .filter(tree => tree.tree_name?.includes(inputFilter) || tree.git_commit_hash?.includes(inputFilter) || tree.patchset_hash?.includes(inputFilter) || tree.git_repository_branch?.includes(inputFilter))
```
Comment (viniciuslacerda0) [inline dashboard/src/components/TreeListingPage/TreeListingPage.tsx:79]: nit: i would just break in multiple lines to make it easier to read

**Review by viniciuslacerda0 — APPROVED** [review]

File: dashboard/src/components/TopBar/TopBar.tsx (hunk 1 above)
Comment (mari1912) [reply dashboard/src/components/TopBar/TopBar.tsx:20]: yes, we will fix later

**Review by lfjnascimento — APPROVED** [review]

**Pull request merged**